src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,67 @@
+        result.add(new Product(6L, "htces4g", "HTC EVO Shift 4G", new BigDecimal(320), usd, 3, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/HTC/HTC%20EVO%20Shift%204G.jpg"));
+        result.add(new Product(7L, "sec901", "Sony Ericsson C901", new BigDecimal(420), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Ericsson%20C901.jpg"));
+        result.add(new Product(8L, "xperiaxz", "Sony Xperia XZ", new BigDecimal(120), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Xperia%20XZ.jpg"));
+        result.add(new Product(9L, "nokia3310", "Nokia 3310", new BigDecimal(70), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Nokia/Nokia%203310.jpg"));
+        result.add(new Product(10L, "palmp", "Palm Pixi", new BigDecimal(170), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Palm/Palm%20Pixi.jpg"));
+        result.add(new Product(11L, "simc56", "Siemens C56", new BigDecimal(70), usd, 20, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C56.jpg"));
+        result.add(new Product(12L, "simc61", "Siemens C61", new BigDecimal(80), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C61.jpg"));
+        result.add(new Product(13L, "simsxg75", "Siemens SXG75", new BigDecimal(150), usd, 40, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg"));
+
+    }
     @Override
     public Product getProduct(Long id) {
-        throw new RuntimeException("Not implemented");
+        return result.stream().filter(x-> x.getId()==id).findFirst().get();
     }
-
+    
+    @Override
+    public boolean getProd(Long id){
+        return result.stream().anyMatch(x->x.getId()==id);
+    }
+    
     @Override
     public List<Product> findProducts() {
-        throw new RuntimeException("Not implemented");
+        List<Product> list = result.stream().filter(x-> x.getPrice()!=null && x.getStock()>0).collect(Collectors.toCollection(ArrayList::new));
+        return list;
     }
 
     @Override
     public void save(Product product) {
-        throw new RuntimeException("Not implemented");
+        if(this.result.stream().anyMatch(x-> x.getId()==product.getId())){
+            throw new IllegalArgumentException("Object's id isn't unic");
+        }
+        else{
+            this.result.add(product);
+        }
     }
 
     @Override
     public void delete(Long id) {
-        throw new RuntimeException("Not implemented");
+        if(this.result.stream().anyMatch(x->x.getId() == id)){
```

Comment:
Can we do the same without if-else block and getting by index?
.filter(...).findFirst().map(...remove product here...).orElseThrow(...)